src/tests/components/IconList.test.ts

```
@@ -0,0 +1,53 @@
+const iconsPromise = vi.hoisted(() => import("../mocks/icon.store.mock"));
+
+beforeEach(() => setupIntersectionObserverMock({ observe: vi.fn() }));
+
+// Mock the stores
+vi.mock("../stores/icon.store", () => iconsPromise);
+
+afterEach(vi.restoreAllMocks);
+
+describe("IconList", () => {
+	it("correctly renders the list of icons in the store", async () => {
+		let icons = new Array(50).fill({ ..._icon.mkEmpty() })
```

ESLint

[error] 'icons' is never reassigned. Use 'const' instead.

(prefer-const)